Xmf2.Common.Extensions/Extensions/DateExtensions.cs

```diff
@@ -54,6 +54,57 @@ public static bool IsSameYear(this DateTime firstDate, DateTime secondDate)
 			return firstDate.Year == secondDate.Year;
 		}
 
+		#region DateTimeOffset
+
+		public static DateTime BeginningOfDay(this DateTimeOffset date) => date.Date;
+
+		public static DateTime EndOfDay(this DateTimeOffset date)
+		{
+			var endOfDay = new DateTime(date.Year, date.Month, date.Day);
+			endOfDay = endOfDay.AddDays(1);
+			endOfDay = endOfDay.AddTicks(-1);
+
+			return endOfDay;
+		}
+
+		public static DateTime BeginningOfMonth(this DateTimeOffset date)
+		{
+			return new DateTime(date.Year, date.Month, 1);
+		}
+
+		public static DateTime BeginningOfWeek(this DateTimeOffset date, DayOfWeek starDayOfWeek = DayOfWeek.Monday)
+		{
+			int diff = (7 + (date.DayOfWeek - starDayOfWeek)) % 7;
+			return date.AddDays(-1 * diff).Date;
+		}
+
+		public static bool IsSameDay(this DateTimeOffset firstDate, DateTimeOffset secondDate)
+		{
+			return firstDate.IsSameMonth(secondDate) && firstDate.Day == secondDate.Day;
+		}
+
+		public static bool IsSameWeek(this DateTimeOffset firstDate, DateTimeOffset secondDate, DayOfWeek startDayOfWeek = DayOfWeek.Monday)
+		{
+			DateTimeFormatInfo dfi = DateTimeFormatInfo.CurrentInfo;
+
+			int firstDateWeek = dfi.Calendar.GetWeekOfYear(firstDate.Date, dfi.CalendarWeekRule, startDayOfWeek);
+			int secondDateWeek = dfi.Calendar.GetWeekOfYear(secondDate.Date, dfi.CalendarWeekRule, startDayOfWeek);
+
+			return firstDateWeek == secondDateWeek;
+		}
+
+		public static bool IsSameMonth(this DateTimeOffset firstDate, DateTimeOffset secondDate)
```

Comment:
Add a overload for every `IsSameXXXX` method with `TimeSpan offset` argument woud be nice. Eg:
		```
public static bool IsSameMonth(this DateTimeOffset firstDate, DateTimeOffset secondDate, TimeSpan offset)
		{
			 return IsSameMonth(firstDate.ToOffset(offset), secondDate.ToOffset(offset))
		}
```